spec/requests/list_controller_spec.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require_relative '../support/assign_allowed_group'
+
+describe ListController do
+
+  before { SiteSetting.assign_enabled = true }
+
+  let(:user) { Fabricate(:active_user) }
+  let(:user2) { Fabricate(:user) }
+  let(:admin) { Fabricate(:admin) }
+  let(:post) { Fabricate(:post) }
+
+  describe 'only allow users from allowed groups' do
+    include_context 'A group that is allowed to assign'
+
+    it 'filters requests where current_user is not member of an allowed group' do
+      sign_in(user)
+      SiteSetting.assign_allowed_on_groups = ''
+
+      get "/topics/group-topics-assigned/#{get_assigned_allowed_group_name}.json"
+      expect(response.status).to eq(403)
+
+      get "/topics/messages-assigned/#{user.username_lower}.json"
+      expect(response.status).to eq(403)
+    end
+
+    it 'as an anon user' do
+      get "/topics/group-topics-assigned/#{get_assigned_allowed_group_name}.json"
+      expect(response.status).to eq(403)
+
+      get "/topics/messages-assigned/#{user.username_lower}.json"
+      expect(response.status).to eq(403)
+    end
+
+    it 'as an admin user' do
+      sign_in(admin)
+      get "/topics/group-topics-assigned/#{get_assigned_allowed_group_name}.json"
+      expect(response.status).to eq(200)
+
+      get "/topics/messages-assigned/#{user.username_lower}.json"
+      expect(response.status).to eq(200)
+    end
+  end
+
+  context '#group_topics_assigned' do
+    include_context 'A group that is allowed to assign'
+
+    fab!(:post1) { Fabricate(:post) }
+    fab!(:post2) { Fabricate(:post) }
+    fab!(:post3) { Fabricate(:post) }
+
+    before do
+      add_to_assign_allowed_group(user)
+
+      freeze_time 1.hour.from_now
+      TopicAssigner.new(post1.topic, user).assign(user)
+
+      freeze_time 1.hour.from_now
+      TopicAssigner.new(post1.topic, user).assign(user2)
+
+      sign_in(user)
```

Comment:
Why are the freeze_time things needed here?

Reply:
I've seen this (here)[https://github.com/discourse/discourse-assign/blob/68bebd77f639e16f1f002ff91d05f46f91ce61d5/spec/requests/assign_controller_spec.rb#L156], so used as it is😅😅

Reply:
Ah I see. In that case, it was testing that the topics were ordered by 'assigned date', so having precise times was important to the test. 

I think we can safely remove it here :)

Reply:
sure, ill remove this right now.